File: android/src/main/kotlin/com/cleveradssolutions/plugin/flutter/util/CASExtensions.kt
package com.cleveradssolutions.plugin.flutter.util

import com.cleversolutions.ads.AdError
import com.cleversolutions.ads.AdImpression

fun AdError.toMap(): Map<String, Any?> {
    return mapOf(
        "code" to code,
        "message" to message
    )
}

fun AdImpression.toMap(): Map<String, Any?> {
    return mapOf(
        "cpm" to this.cpm,
        "priceAccuracy" to this.priceAccuracy,
        "adType" to this.adType.ordinal,
        "networkName" to this.network,
        "versionInfo" to this.versionInfo,
        "identifier" to this.identifier,
        "impressionDepth" to this.impressionDepth,
        "lifetimeRevenue" to this.lifetimeRevenue,
        "creativeIdentifier" to this.creativeIdentifier
    )
}

File: ios/Classes/Util/CASExtensions.swift
import CleverAdsSolutions
import Foundation

extension AdError {
    func toDict() -> [String: Any?] {
        return [
            "code": code,
            "message": description,
        ]
    }
}

extension CASImpression {
    func toDict() -> [String: Any?] {
        return [
            "cpm": cpm,
            "priceAccuracy": priceAccuracy.rawValue,
            "adType": adType.rawValue,
            "networkName": network,
            "versionInfo": versionInfo,
            "identifier": identifier,
            "impressionDepth": impressionDepth,
            "lifetimeRevenue": lifetimeRevenue,
            "creativeIdentifier": creativeIdentifier,
        ]
    }
}
